tests/comm/test_quantized_allreduce.py

```
@@ -0,0 +1,273 @@
+
+
+@pytest.mark.skipif(
+    torch.cuda.device_count() < 2,
+    reason="Tests require at least 2 CUDA devices",
+)
+@pytest.mark.skipif(
+    get_compute_capability(torch.device("cuda:0"))[0] < 9,
+    reason="Requires SM90+ (Hopper or later)",
+)
+def test_quantized_allreduce_correctness():
+    port = random.randint(30000, 60000)
```

Ruff (0.15.18)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)